### PR TITLE
V2 mod upgrade

### PR DESCRIPTION
**Pull request by prestonvasquez**
## Summary

Update the mod and self import paths with v2.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,7 +26,7 @@ build-tests:
 
 .PHONY: build-compile-check
 build-compile-check:
-	etc/compile_check.sh
+	#etc/compile_check.sh
```
Comment (prestonvasquez) [inline Makefile:29]: TODO: before merge, uncomment

File: internal/cmd/compilecheck/go.mod [hunk 2]
```diff
@@ -6,8 +6,7 @@ replace go.mongodb.org/mongo-driver => ../../../
 
 // Note that the Go driver version is replaced with the local Go driver code by
 // the replace directive above.
-//require go.mongodb.org/mongo-driver
-require go.mongodb.org/mongo-driver v1.11.7
+//require go.mongodb.org/mongo-driver/v2 v2.0.0
```
Comment (prestonvasquez) [inline internal/cmd/compilecheck/go.mod:9]: TODO: before merge uncomment and update with alpha

Comment (blink1073) [reply internal/cmd/compilecheck/go.mod:9]: This is ready: https://github.com/mongodb/mongo-go-driver/actions/runs/10098917125

Comment (prestonvasquez) [reply internal/cmd/compilecheck/go.mod]: On further investigation, I think we will have to wait for a release that contains `v2` in the import path in order for the compile check to work. I suggest releasing alpha2 after this is merged, and then we can create a followup PR to fix the compile check.

@matthewdale Do you have thoughts?

Comment (prestonvasquez) [reply internal/cmd/compilecheck/go.mod]: The same goes for all tests in internal/cmd that are modules.

Comment (matthewdale) [reply internal/cmd/compilecheck/go.mod]: Doing this in a follow-up PR seems fine.

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

**Pull request merged**